packages/react-native-bridge/ios/GutenbergLayouts.swift

```diff
@@ -0,0 +1,140 @@
+import Foundation
+
+public struct GutenbergPageLayouts {
+    public let layouts: [GutenbergLayout]
+    public let categories: [GutenbergLayoutCategory]
+
+    /// Contains a map of layouts based on their Category slug
+    private let groupedLayouts: [String: [GutenbergLayout]]
+
+
+    public init(layouts: [GutenbergLayout], categories: [GutenbergLayoutCategory]) {
+        self.layouts = layouts
+        self.categories = categories
+        groupedLayouts = GutenbergPageLayouts.groupLayouts(layouts)
+    }
+
+    static func groupLayouts(_ layouts: [GutenbergLayout]) -> [String:[GutenbergLayout]] {
+
+        var groupedLayouts = [String:[GutenbergLayout]]()
+
+        layouts.forEach { (layout) in
+            var group = groupedLayouts[layout.slug] ?? [GutenbergLayout]()
+            group.append(layout)
+            groupedLayouts.updateValue(group, forKey: layout.slug)
+        }
+
+        return groupedLayouts
+    }
+
+    public func layouts(forCategory slug: String) -> [GutenbergLayout] {
+        return groupedLayouts[slug] ?? []
+    }
+}
+
+public struct GutenbergLayout {
+    public let slug: String
+    public let title: String
+    public let preview: String
+    public let categories: [GutenbergLayoutCategory]
```

Comment:
Not sure why this is an array 🤔 Are we aware of any use case where a layout appears under different categories?

Reply:
I'm not sure if we would expect multiple categories for a layout or not. My goal with the `structs` `GutenbergPageLayouts`, `GutenbergLayout`, and `GutenbergLayoutCategory` was to mirror the syntax of the response that comes from the API to make parsing those a bit easier and to prevent having one struct for the API and another for the UI. 

> Otherwise we'll need to search in GutenbergLayout instances to get layouts for a specific category.

To prevent having to do this I added [groupedLayouts](https://github.com/WordPress/gutenberg/pull/24027/files/04aba7bf4bb70e0c8bc759e0f4b5c924f666210e#diff-527e778ee2c6d383dea1df847278893eR8) to so that we only need to organize the layouts into categories once. Since it's a small `n` I figured there wouldn't be a noticeable performance hit. 

<details>
<summary>Sample API Response</summary>

```
{
    "layouts": [
        {
            "slug": "about-2",
            "title": "About",
            "content": "...",
            "categories": [
                {
                    "slug": "about",
                    "title": "About",
                    "description": "About pages",
                    "emoji": "👋"
                }
            ],
            "preview": "https://headstartdata.files.wordpress.com/2020/01/about-2.png?w=200&zoom=1"
        },
        ...
    ],
    "categories": [
        {
            "slug": "about",
            "title": "About",
            "description": "About pages",
            "emoji": "👋"
        }
        ...
    ]
}
```

</details>

Reply:
> mirror the syntax of the response that comes from the API

OK thanks, LGTM then!